plugin/src/sbt-test/scalafmt-sbt/sbt/test

```diff
@@ -94,16 +94,38 @@ $ touch p14/src/main/scala/Test.scala
 
 > p15/scalafmt
 > p15/scalafmtCheck
-$ copy-file .scalafmt.conf .scalafmt15.conf
 # prevent reading the source (without changing the mtime) to detect actual/uncached invocation of scalafmt
 $ exec chmod 000 p15/src/main/scala/Test.scala
+######## full run expected when config was updated and a source file added since last run
+$ copy-file .scalafmt.conf .scalafmt15.conf
+$ copy-file changes/bad.scala p15/src/main/scala/Test2.scala
+-> p15/scalafmt
+-> p15/scalafmtCheck
+######## full run expected when only config was updated since last run
+$ delete p15/src/main/scala/Test2.scala
 -> p15/scalafmt
 -> p15/scalafmtCheck
 
 > p16/scalafmt
 > p16/scalafmtCheck
-> set p16/scalafmtConfig := file(".scalafmt.conf")
 # prevent reading the source (without changing the mtime) to detect actual/uncached invocation of scalafmt
 $ exec chmod 000 p16/src/main/scala/Test.scala
+######## full run expected when config was updated and a source file added since last run
+> set p16/scalafmtConfig := file(".scalafmt.conf")
+$ copy-file changes/bad.scala p16/src/main/scala/Test2.scala
+-> p16/scalafmt
+-> p16/scalafmtCheck
+######## full run expected when only config was updated since last run
+$ delete p16/src/main/scala/Test2.scala
 -> p16/scalafmt
 -> p16/scalafmtCheck
+
+$ copy-file changes/bad.scala p17/src/main/scala/Test2.scala
+-> p17/scalafmtCheck
+# prevent reading the source (without changing the mtime) to detect actual/uncached invocation of scalafmt
+$ exec chmod 000 p17/src/main/scala/Test.scala
+######## incremental checking expected when all previous validations failed on some sources only
+> p17/test:scalafmt
+$ copy-file changes/bad.scala p17/src/main/scala/Test2.scala
+-> p17/scalafmtCheck
```

Comment:
It seems that this test is no longer relevant: there is no source in `test` anymore, so the 2 lines above are no-op, and this does not verify the incrementality of `scalafmtCheck`. That process got lost in https://github.com/scalameta/sbt-scalafmt/pull/46/commits/1944e56c87856cee655032e5a71206ecba85d7e4 from what I can see.

The original scripted was formatting the content of `bad.scala` "outside" and copying it back to the same place (to simulate a user update, rather than exercising the project/config tasks themselves which would have biased the test), expecting the check here to run successfully on that updated file without re-running on the `Test.scala` bombed above. I thought using `test/scala/Test2.scala` & `test:scalafmt` as a sandbox was better than checking-in 2 files (one being unformatted, the other one formatted) to show the intent, but it was obviously not clear! What do you think?

Reply:
ah. I totally did not see `test:` in `> p17/test:scalafmt`. I think it's better to check in two files.